emperor/flat_plots.py

```diff
@@ -0,0 +1,346 @@
+from matplotlib import use
+use('Agg')  # noqa
+import matplotlib.pyplot as plt
+import numpy as np
+import seaborn as sns
+
+
+def plot_3x3(ordination, offset, mapping, field, colors, null_kw=None,
+             focus_kw=None):
+    """Construct a 3 x 3 plot of PC combinations
+
+    Where X is a PCn / PCm plot, _ is a blank spot, and L is the legend.
+
+    Offset defines the starting offset for the rows. If zero, the layout
+    is:
+
+    PC1v2 PC1v3 PC1v4 BLANK
+    PC2v3 PC2v4 PC2v5 Legend
+    PC3v4 PC3v5 PC3v6 BLANK
+
+    Parameters
+    ----------
+    ordination : OrdinationResults
+        An OrdinationResults object
+    offset : int
+        The starting ordination axis (e.g., 0 in the above example as 0 maps
+        to PC1).
+    mapping : DataFrame
+        DataFrame object with the metadata associated to the samples in the
+        ``ordination`` object, should have an index set and it should match the
+        identifiers in the ``ordination`` object.
+    field : str
+        The field within the mapping file to use for coloring.
+    colors : dict or str
+        If a dict, it is expected to provide a mapping of category value to
+        a color; entries not in the mapping will be considered "null". If a
+        str, the value is interpreted as the name of a color map and assumes
+        a gradient is to be painted.
+    null_kw : dict, optional
+        Parameters to use for plotting and painting null or unknown values.
+    focus_kw : dict, optional
+        Parameters to use for plotting and painting values which are not
+        null and in the field of interest.
+    """
```

Comment:
It would be nice to add a note that there is an intersection operation occurring - that only the samples contained in both the mapping file and the ordination results will be considered.